core/shared/src/main/scala/zio/clock/package.scala

```diff
@@ -45,11 +47,13 @@ package object clock {
             Left(UIO.effectTotal(canceler()))
           }
 
-        def currentDateTime: ZIO[Any, Nothing, OffsetDateTime] =
-          for {
-            millis <- currentTime(TimeUnit.MILLISECONDS)
-            zone   <- ZIO.effectTotal(ZoneId.systemDefault)
-          } yield OffsetDateTime.ofInstant(Instant.ofEpochMilli(millis), zone)
+        def currentDateTime: IO[DateTimeException, OffsetDateTime] =
+          (for {
```

Comment:
Could define an intermediate value here if you wanted to avoid the awkward parentheses around the for comprehension.

Reply:
Yeah, makes sense, I'll do that